File: examples/indexer/java/java-test/src/main/java/com/algorand/javatest/indexer/SearchAssetsName.java
// SearchAssetsName.java
// requires java-algorand-sdk 1.4.0 or higher (see pom.xml)
package com.algorand.javatest.indexer;

import com.algorand.algosdk.v2.client.common.IndexerClient;
import com.algorand.algosdk.v2.client.common.Client;
import org.json.JSONObject;

public class SearchAssetsName {
    public Client indexerInstance = null;
    // utility function to connect to a node
    private Client connectToNetwork(){
        final String INDEXER_API_ADDR = "localhost";
        final int INDEXER_API_PORT = 8980;       
        IndexerClient indexerClient = new IndexerClient(INDEXER_API_ADDR, INDEXER_API_PORT); 
        return indexerClient;
    }
    public static void main(String args[]) throws Exception {
        SearchAssetsName ex = new SearchAssetsName();
        IndexerClient indexerClientInstance = (IndexerClient)ex.connectToNetwork();
        String name = "DevDocsCoin";        
        String response = indexerClientInstance.searchForAssets()
                        .name(name).execute().toString();
        JSONObject jsonObj = new JSONObject(response.toString());
        System.out.println("Asset Info for Name: " + jsonObj.toString(2)); // pretty print json
    }
 }

File: examples/indexer/indexer/javascript/searchAssetsName.js
// SearchAssetName.js
// requires algosdk@1.6.1 or higher 
// verify installed version
// npm list algosdk

const algosdk = require('algosdk');
const indexer_token = "";
const indexer_server = "http://localhost";
const indexer_port = 8980;

// Instantiate the indexer client wrapper
let indexerClient = new algosdk.Indexer(indexer_token, indexer_server, indexer_port);

(async () => {
    let name = "DevDocsCoin";
    let assetInfo = await indexerClient.searchForAssets()
        .name(name).do();
    console.log("Information for Asset Name: " + JSON.stringify(assetInfo, undefined, 2));
})().catch(e => {
    console.log(e);
    console.trace();
});